commons/src/main/java/io/aiven/kafka/connect/common/source/input/ByteArrayTransformer.java

```diff
@@ -18,33 +18,38 @@
 
 import java.io.IOException;
 import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
-import java.util.Map;
 import java.util.function.Consumer;
 
 import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.data.SchemaAndValue;
 
+import io.aiven.kafka.connect.common.OffsetManager;
+
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.io.function.IOSupplier;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class ByteArrayTransformer extends Transformer<byte[]> {
+/**
+ * A transformer that reads plain bytes from the input stream.
+ */
+public class ByteArrayTransformer extends Transformer {
+    /** The logger for this transformer */
     private static final Logger LOGGER = LoggerFactory.getLogger(ByteArrayTransformer.class);
-
+    /** The maximum record size this transform will read. If more data is sent then the record is ignored. */
     private static final int MAX_BUFFER_SIZE = 4096;
```

Comment:
I feel like this should be configureable in the CommonSourceConfig but I think we can do it in another PR, can you create a ticket for that?